(x) check first launch

Gzort opened the pull request:
### [feat]: Add First launch checking via SharedPrefs. 

- Add **FirstLaunchCheckerRepository.kt** in r**u.itmo.domain.repositories**
- Add **CheckFirstLaunchUseCase.kt** and **CompleteFirstLaunchUseCase.kt** in **ru.itmo.domain.usecases**
- Create package **ru.itmo.data.prefs** and add **FirstLaunchCheckerRepositoryImpl.kt** there
- Add First launch checking in **InitActivity.kt** and **WelcomeActivity.kt** via created UseCases

Gzort committed:
[feat]: Add First launch checking.
Add FirstLaunchCheckerRepository.kt in ru.itmo.domain.repositories
Add CheckFirstLaunchUseCase.kt and CompleteFirstLaunchUseCase.kt in ru.itmo.domain.usecases
Create package prefs in ru.itmo.data and add FirstLaunchCheckerRepositoryImpl.kt there
Add First launch checking in InitActivity.kt and WelcomeActivity.kt

Gzort commented:
Change target branch to develop

DanilB01 commented on app/src/main/java/ru/itmo/travelhelper/screens/InitActivity.kt:
```diff
@@ -3,20 +3,21 @@ package ru.itmo.travelhelper.screens
 
-        val isFirstLaunching = true //get bool value if first launching true/false
+        val isFirstLaunching =
+            CheckFirstLaunchUseCase(FirstLaunchCheckerRepositoryImpl(this@InitActivity)).execute()
```
Согласно принципам MVP-архитектуры, использовать UseCase необходимо в классе презентора - InitPresentor. Следует вынести туда туда обращение к юзкейсу (для примера посмотри активити с HotelPresentor). Также не забудь про интерфейс View (также см. пример экрана с отелями)

DanilB01 commented on data/src/main/java/ru/itmo/data/prefs/FirstLaunchCheckerRepositoryImpl.kt:
```diff
@@ -0,0 +1,22 @@
+
+class FirstLaunchCheckerRepositoryImpl(context: Context) : FirstLaunchCheckerRepository {
+
+    private val preferences: SharedPreferences =
```
Работа с SharedPrefs - это работа с данными, а работа с данными описывается в модуле data, который за это и отвечает. Таким образом, чтобы все соответствовало выбранной архитектуре нашего проекта, нужно сделать по аналогии с репозиториями, то есть:
1) завести интерфейс SharedPrefs в модуле domain, в котором описать все методы, которые реализуются в префсах. В нашем случае - интерфейс должен содержать метод setFirstLaunchCompleted и функцию isFirstLaunch.
2) реализацию этого интерфейса необходимо сделать в модуле data. там уже создать приватную переменную префсов, а также реализовать методы интерфейса из первого пункта.
3) в параметрах юзкейса в конструкторе добавить параметр, отвечающий за префсы (вместо передаваемого контекста), причем юзкейс должен ожидать параметр типа интерфейса из пункта 1
4) в презенторе создавать и передавать юзкейсу экземпляр префсов.
Примерно посмотреть, как это выглядит, можно в примере реализации отелей.

Gzort committed:
[fix]: Moved UseCase using to Presentors
Renamed FirstLaunchCheckerRepository.kt to LocalStorage.kt
Renamed FirstLaunchCheckerRepositoryImpl.kt to LocalStorageImpl.kt

Gzort committed:
[fix]: Remove unused import
